contrib/externs/chrome_extensions.js

```diff
@@ -11016,3 +11016,72 @@ chrome.downloads.DeterminingFilenameEvent.prototype.hasListeners =
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/** @type {string} */
+chrome.declarativeContent.PageStateMatcherInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.ShowPageActionInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.SetIconInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.RequestContentScriptInstanceType;
+
+
+/**
+ * @typedef {!{
+ *   pageUrl: (!UrlFilter|undefined),
+ *   css: (!Array<string>|undefined),
+ *   isBookmarked: (boolean|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcher;
+
+
+/**
+ * @constructor
```

Comment:
As I understand it, the compiler should complain if you do any of these.

```javascript
/** @type {!Array<!chrome.declarativeContent.PageStateMatcher>} */
const rules = [
    new chrome.declarativeContent.PageStateMatcher(15),  // wrong argument type
    chrome.declarativeContent.PageStatematcher({  // missing 'new'
      pageUrl: { hostEquals: 'www.google.com', schemes: ['https'] },
      css: ["input[type='password']"]
    }),
    {  // bare literal where you needed to pass a constructed object.
      pageUrl: { hostEquals: 'www.google.com', schemes: ['https'] },
      css: ["input[type='password']"]
    } ];
```
If you define `PageStateMatcher` as a typedef the compiler won't catch any of these.
It won't catch the first 2 because the old type system basically treats the typedef'ed symbol as an unknown type when it is used in code instead of JSDoc, so it doesn't check it at all.
It won't catch the last one because the typedef actually says that the bare literal is a `PageStateMatcher`.

Does that clarify things at all?